### PR TITLE
RHAIENG-2777: refactor(.tekton): drop -odh-main- infix from base image pipeline filenames

### LINKED ISSUE #3510: Fix PipelineRun naming so trigger-pac-build targets the correct pipeline
## Context

The `trigger-pac-build` annotation works by having the build-service POST to PaC's `/incoming` endpoint with `pipelinerun: <component-name>-on-push`. PaC then searches `.tekton/` for a PipelineRun whose `metadata.name` matches that string.

Currently there are two mismatches:

**Main-branch push files** (`*-odh-main-push.yaml`):
- Component: `odh-pipeline-runtime-minimal-cpu-py312-ubi9` (branch=main)
- Build-service looks for: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push`
- Actual `metadata.name`: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-odh-main-push`
- **Mismatch**: `-odh-main-push` vs `-on-push`

**Stable-branch push files** (`*-push.yaml`):
- Component: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci` (branch=stable)
- Build-service looks for: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci-on-push`
- Actual `metadata.name`: `odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push`
- **Mismatch**: component has `-ci` but pipelinerun name doesn't

**Base image additional complication**: some base image components have version-agnostic names (e.g. `odh-base-image-cuda-py312-c9s`) but the files have version-specific names (e.g. `odh-base-image-cuda-12-9-py312-c9s-odh-main-push`). Multiple files map to one component — these can't be fixed by renaming alone.

### 1. Rename stable push pipeline `metadata.name` fields (19 files)

Change from `<base-name>-on-push` to `<base-name>-ci-on-push` to match the `-ci` component name.

Example: in `odh-pipeline-runtime-minimal-cpu-py312-ubi9-push.yaml`:
```yaml
# before
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push
# after
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-ci-on-push
```

**Files** (all `*-push.yaml` excluding `*-odh-main-push.yaml`):
- `odh-pipeline-runtime-{datascience-cpu,minimal-cpu,pytorch-cuda,pytorch-llmcompressor-cuda,pytorch-rocm,tensorflow-cuda,tensorflow-rocm}-py312-ubi9-push.yaml` (7)
- `odh-workbench-{codeserver-datascience-cpu,jupyter-datascience-cpu,jupyter-minimal-cpu,jupyter-minimal-cuda,jupyter-minimal-rocm,jupyter-pytorch-cuda,jupyter-pytorch-llmcompressor-cuda,jupyter-pytorch-rocm,jupyter-tensorflow-cuda,jupyter-tensorflow-rocm,jupyter-trustyai-cpu}-py312-ubi9-push.yaml` (11)
- No rstudio files (removed)

### 2. Rename main push pipeline `metadata.name` fields (non-base-image, ~18 files)

Change from `<base-name>-odh-main-push` to `<base-name>-on-push` to match the component name pattern build-service expects.

Example: in `odh-pipeline-runtime-minimal-cpu-py312-ubi9-odh-main-push.yaml`:
```yaml
# before
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-odh-main-push
# after  
name: odh-pipeline-runtime-minimal-cpu-py312-ubi9-on-push
```

### 3. Base image files — skip or defer

Base image components have version-agnostic names (`odh-base-image-cuda-py312-c9s`) but version-specific pipeline files (`cuda-12-9`, `cuda-13-0`). Multiple files map to one component, so `trigger-pac-build` can't know which to trigger. Skip these — they only build from main and are triggered by pathChanged() on real pushes.

## Verification

```bash
# Check that each component's expected pipelinerun name matches the metadata.name in .tekton/
# For stable:
for f in .tekton/*-push.yaml; do
  [[ "$f" == *-odh-main-push.yaml ]] && continue
  name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  [[ "$name" != "$expected" ]] && echo "MISMATCH: $f name=$name expected=$expected"
done

# For main (non-base-image):
for f in .tekton/*-odh-main-push.yaml; do
  [[ "$f" == *base-image* ]] && continue
  name=$(grep '^  name:' "$f" | head -1 | awk '{print $2}')
  comp=$(grep 'appstudio.openshift.io/component:' "$f" | head -1 | awk '{print $2}')
  expected="${comp}-on-push"
  [[ "$name" != "$expected" ]] && echo "MISMATCH: $f name=$name expected=$expected"
done
```

After renaming, `trigger-pac-build` should produce a matching PipelineRun for both main and stable components.

Also run the existing test suite:
```bash
./uv run pytest tests/test_main.py -x
```

### LINKED ISSUE #3521: refactor(.tekton): drop -odh-main- infix from base image pipeline filenames
## Problem

The 16 base image pipeline files in `.tekton/` still use the `-odh-main-` infix in their filenames (e.g. `odh-base-image-cuda-12-9-py312-c9s-odh-main-push.yaml`). This infix was introduced to avoid filename collisions between main and stable branch pipelines, but **base images only build from `main`** — there are no `-ci` / stable counterparts. The infix is unnecessary and inconsistent with the runtime/workbench files which have already been cleaned up.

## Analysis

Base image component mapping:

| Component | Files | Can use `<component>-{push,pull-request}.yaml`? |
|---|---|---|
| `odh-base-image-cpu-py312-c9s` | 2 (push + PR) | Yes — 1:1 mapping |
| `odh-base-image-cpu-py312-ubi9` | 2 | Yes — 1:1 |
| `odh-base-image-cuda-py312-c9s` | 4 (cuda-12-9 + cuda-13-0, push + PR each) | No — 2 version-specific files share 1 component |
| `odh-base-image-cuda-py312-ubi9` | 2 (cuda-12-9 only) | Yes — 1:1 |
| `odh-base-image-rocm-py312-c9s` | 4 (rocm-6-4 + rocm-7-1) | No — same issue |
| `odh-base-image-rocm-py312-ubi9` | 2 (rocm-6-4 only) | Yes — 1:1 |

For the multi-version files (cuda c9s, rocm c9s), we can't use `<component>-push.yaml` because two files would collide. But we can still drop `-odh-main-` and use the version-specific name directly: `odh-base-image-cuda-12-9-py312-c9s-push.yaml` instead of `odh-base-image-cuda-12-9-py312-c9s-odh-main-push.yaml`.

## Proposed steps

1. `git mv` all 16 files, removing `-odh-main-` from filenames
2. Update CEL `pathChanged()` expressions in each file to reference the new filename
3. Verify no duplicate names in `metadata.name` (the multi-version files already have unique version-specific names)
4. Run tests

The `metadata.name` fields don't need to change — they were already fixed in PR #3511.

## References

- [PR #3511](https://github.com/opendatahub-io/notebooks/pull/3511) — fixed `metadata.name` and renamed runtime/workbench files
- `.tekton/README-odh.md` "Base images" section — documents the version-specific filename constraint
- [Slack thread](https://redhat-internal.slack.com/archives/C0961HQ858Q/p1768219590576299) — original `-odh-main-` naming discussion

### PR DESCRIPTION
**Pull request by jiridanek**
Fixes https://github.com/opendatahub-io/notebooks/issues/3510
https://redhat.atlassian.net/browse/RHAIENG-2777

## Summary

- Rename all 16 base image pipeline files in `.tekton/` to drop the unnecessary `-odh-main-` infix from filenames
- Update CEL `pathChanged()` self-references in each file to match the new filename

Base images only build from `main` — there are no `-ci` / stable counterparts — so the `-odh-main-` infix serves no purpose and is inconsistent with the runtime/workbench files which were already cleaned up in #3511.

Example rename:
```
odh-base-image-cuda-12-9-py312-c9s-odh-main-push.yaml
→ odh-base-image-cuda-12-9-py312-c9s-push.yaml
```

## History

The `-odh-main-` infix was originally introduced in #3080 to avoid filename collisions between main and stable branch pipelines. With the `-ci` suffix convention making stable filenames unique, the collision concern no longer applies.

PR #3511 unified the naming for runtime and workbench pipelines and fixed `metadata.name` fields, but deliberately left base images for a separate PR because multi-version files (cuda-12-9 / cuda-13-0, rocm-6-4 / rocm-7-1) share a single component and can't use the simple `<component>-push.yaml` pattern. This PR completes that cleanup by using the version-specific name directly.

## What's NOT changed

- `metadata.name` fields — left as-is per #3511 scope
- `pipelineRef.name: multiarch-odh-main-combined-pipeline` — references the shared pipeline definition, not a filename
- `multiarch-odh-main-combined-pipeline.yaml` — not a base image pipeline file

Closes #3521

## Test plan

- [x] Verify no `*-odh-main-*` base image files remain
- [x] Verify all `pathChanged()` expressions reference the new filenames
- [x] Verify `metadata.name` fields are unchanged

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **Chores**
  * Updated CI/CD pipeline trigger conditions across multiple base-image build workflows (CPU, CUDA, ROCm). Triggers now reference the correct workflow files so pull-request and push events run the intended pipelines; no other pipeline parameters, tasks, or runtime behavior were changed.

### CODE REVIEW

**Review by atheo89 — APPROVED** [review]

File: .tekton/odh-base-image-cpu-py312-c9s-pull-request.yaml
Comment (jiridanek) [inline .tekton/odh-base-image-cpu-py312-c9s-pull-request.yaml:1]: @coderabbitai
                        
  Pure housekeeping PR: renames 16 .tekton/ base image pipeline YAML files to remove the -odh-main- infix, and updates the CEL pathChanged() self-references in each file to match. No logic, parameters, or pipeline behavior changes.                                                                                   
   
  Correctness                                                                                                                                                                                                                                                                                                             
                        
  All 16 renames are consistent and correct. Each file:                                                                                                                                                                                                                                                                   
  - File rename: drops -odh-main- from filename
  - CEL expression: pathChanged() string updated to match the new filename                                                                                                                                                                                                                                                
                                                                          
  Verified:                                                                                                                                                                                                                                                                                                               
  - Zero files with the old -odh-main- naming pattern remain in .tekton/                                                                                                                                                                                                                                                  
  - No stale pathChanged() references pointing to old filenames                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                          
  Intentionally Unchanged (per PR description)                                                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                                                          
  The following still contain odh-main and are correctly left alone:                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                          
  ```
┌──────────────────┬──────────────────────────────────────────────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐                                                                                                              
  │       What       │                     Example                      │                                                               Why                                                                │
  ├──────────────────┼──────────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤                                                                                                              
  │ metadata.name    │ odh-base-image-cuda-12-9-py312-c9s-odh-main-push │ Deliberately deferred (per #3511 scoping). Changing these affects Konflux/PaC pipeline identity and needs separate coordination. │
  ├──────────────────┼──────────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤                                                                                                              
  │ pipelineRef.name │ multiarch-odh-main-combined-pipeline             │ References a shared pipeline definition, not a filename.                                                                         │                                                                                                              
  ├──────────────────┼──────────────────────────────────────────────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤                                                                                                              
  │ README-odh.md    │ Historical note about the infix convention       │ Documents history, not an active reference.                                                                                      │                                                                                                              
  └──────────────────┴──────────────────────────────────────────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘                                                                                                              
```

  Observations                                                                                                                                                                                                                                                                                                            
                        
  1. CPU files already had metadata.name cleaned up (e.g., odh-base-image-cpu-py312-c9s-on-push) while CUDA/ROCm files still carry -odh-main- in their metadata.name. This is the inconsistency that a follow-up PR would address. The PR description acknowledges this scope boundary clearly.                           
  2. No risk to running pipelines. PaC matches pipelines by metadata.name, not filename. Renaming files only affects which pathChanged() paths trigger the pipeline — and those self-references are correctly updated.
  3. Single commit, clean history. Easy to revert if needed.                                                                                                                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                          
  Verdict                                                                                                                                                                                                                                                                                                                 
                                                                                                                                                                                                                                                                                                                          
  Looks good. Mechanical rename with correct self-reference updates. The scope is well-defined and the exclusions are documented. No issues found.

**Comment by jiridanek** [comment]
## Konflux check failures analysis

All 8 Konflux pipeline failures are **infrastructure-related**, not caused by the PR changes.

### Root cause

The `stone-prd-rh01` cluster was under heavy load, causing **resource quota evaluation timeouts** in the `open-data-hub-tenant` namespace. When a new pod is created, the API server must evaluate the ResourceQuota admission check. Under cluster congestion, this evaluation itself times out — before the pod ever reaches the scheduler.

The error from the failed TaskRuns:
```
failed to create task run pod "...": Internal error occurred: resource quota evaluation timed out
```

### Per-pipeline breakdown

| Pipeline | Timeout | Failed Task(s) | Failure Mode |
|----------|---------|----------------|--------------|
| cpu-py312-c9s | 6h | `ecosystem-cert-preflight-checks` (×3) | 1× PodCreationFailed, 2× cancelled |
| cpu-py312-ubi9 | 6h | `ecosystem-cert-preflight-checks` (×1) | cancelled |
| cuda-12-9-py312-c9s | 6h | `ecosystem-cert-preflight-checks` (×2), `clair-scan` (×2) | 1× PodCreationFailed, 3× cancelled |
| cuda-12-9-py312-ubi9 | 6h | `ecosystem-cert-preflight-checks` (×2), `clair-scan` (×1) | 2× PodCreationFailed, 1× cancelled |
| cuda-13-0-py312-c9s | **3h** | `build-images` (×1) | cancelled (build still running at 3h timeout) |
| rocm-6-4-py312-c9s | 6h | `ecosystem-cert-preflight-checks` (×1), `clair-scan` (×1) | cancelled |
| rocm-6-4-py312-ubi9 | 6h | `ecosystem-cert-preflight-checks` (×1) | cancelled |
| rocm-7-1-py312-c9s | 6h | `ecosystem-cert-preflight-checks` (×1) | cancelled |

### Key observations

- **All `build-images` tasks succeeded** in 7 of 8 pipelines — the builds themselves are fine
- The file renames and `pathChanged()` updates worked correctly (pipelines triggered as expected)
- The cuda-13-0 variant has a 3h pipeline timeout (vs 6h for the rest), which is why the build itself got cancelled mid-flight

A re-trigger should pass when the cluster is less busy.
